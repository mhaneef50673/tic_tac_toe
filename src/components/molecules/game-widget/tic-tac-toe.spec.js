import React from "react";
import { mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";
import TicTacToeBoard from "./tic-tac-toe-board";

describe("TicTacToeBoard Component", () => {
  it("should render the component", () => {
    const tree = mount(
      <TicTacToeBoard
        noOfRows={3}
        noOfColumns={3}
        squares={[null, null, null, null, null, null, null, null, null]}
        squareClickHanlder={() => {}}
        winningSquares={[]}
      />
    );
    expect(mountToJson(tree)).toMatchSnapshot();
  });
});
