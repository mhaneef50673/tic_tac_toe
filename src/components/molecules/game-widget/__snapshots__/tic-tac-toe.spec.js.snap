// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToeBoard Component should render the component 1`] = `
<TicTacToeBoard
  noOfColumns={3}
  noOfRows={3}
  squareClickHanlder={[Function]}
  squares={
    Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ]
  }
  winningSquares={Array []}
>
  <div>
    <div
      className="square-row-container"
      key="0"
    >
      <BoardSquare
        isWinning={false}
        key="0"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
      <BoardSquare
        isWinning={false}
        key="1"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
      <BoardSquare
        isWinning={false}
        key="2"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
    </div>
    <div
      className="square-row-container"
      key="1"
    >
      <BoardSquare
        isWinning={false}
        key="3"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
      <BoardSquare
        isWinning={false}
        key="4"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
      <BoardSquare
        isWinning={false}
        key="5"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
    </div>
    <div
      className="square-row-container"
      key="2"
    >
      <BoardSquare
        isWinning={false}
        key="6"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
      <BoardSquare
        isWinning={false}
        key="7"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
      <BoardSquare
        isWinning={false}
        key="8"
        squareClickHanlder={[Function]}
        squareValue={null}
      >
        <button
          className="board-square "
          onClick={[Function]}
        />
      </BoardSquare>
    </div>
  </div>
</TicTacToeBoard>
`;
